### PR TITLE
docs: Add contributor-guide

### PR DESCRIPTION
**Pull request by Xuanwo**
Signed-off-by: Xuanwo <github@xuanwo.io>

### CODE REVIEW

File: docs/general/contributor-guide.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+git remote add upstream git@github.com:beyondstorage/go-storage.git
+```
+
+### Choose an issue
+
+1. Please choose your target issue. If it is a new issue discovered or a new function enhancement to offer, please create an issue and set the right label for it.
+1. After choosing the relevant issue, please reply with a deadline to indicate that you are working on it.
+1. Depends on your working on projects, please ping different groups to give you feedback about your PR or issue in time.
+
+For go-storage related projects (including `go-storage` and `go-storage-*`):
```
Comment (JinnyYi) [inline docs/general/contributor-guide.md:37]: `go-storage-*` -> `go-service-*`?

File: docs/general/contributor-guide.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+git add modified-file-names
+git commit -m 'commit message'
+git push origin <IssueNumber>
+```
+
+### Submit PR
+
+1. Send a pull request to the master branch.
+1. The mentor will do a code review before discussing some details (including the design, the implementation, and the performance) with you.
+1. Also make sure that the pull request title has a semantic prefix like `fix:` or `feat:` or any other [conventional commit types](https://github.com/commitizen/conventional-commit-types/blob/master/index.json).
```
Comment (JinnyYi) [inline docs/general/contributor-guide.md:76]: How about swapping the order of 2 and 3? Does the PR need to add `fixes #IssueNo` or `closes #IssueNo` to close the issue?

File: docs/general/contributor-guide.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+- `@beyond-tp-maintainer`
+- `@beyond-tp-committer`
+- `@beyond-tp-reviewer`
+
+For other projects, there are no dedicated teams responsible for, our community members will try our best.
+
+### Create your branch
+
+Switch to your forked master branch, pull codes from upstream, then create a new branch.
+
```
Comment (Prnyself) [inline docs/general/contributor-guide.md:54]:
```suggestion
Notice: **Every time you try to fix an issue, you should update the upstream and checkout from the master branch to make sure your code is up to date.**
```

File: docs/general/contributor-guide.md [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+### Coding
+
+1. Please obey the Code of Conduct during the process of development and finish the check before submitting the pull request.
+1. Then push codes to your fork repo.
+
+```shell
+git add modified-file-names
+git commit -m 'commit message'
+git push origin <IssueNumber>
```
Comment (Prnyself) [inline docs/general/contributor-guide.md:69]:
```suggestion
git push origin <IssueNo>
```

**Comment by Xuanwo** [comment]
@Prnyself @JinnyYi Please review again.

File: docs/general/contributor-guide.md [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+- `@beyond-tp-maintainer`
+- `@beyond-tp-committer`
+- `@beyond-tp-reviewer`
+
+For other projects, there are no dedicated teams responsible for, our community members will try our best.
+
+### Create your branch
+
+Switch to your forked master branch, pull codes from upstream, then create a new branch.
+Notice: **Every time you try to fix an issue, you should update the upstream and checkout from the master branch to make sure your code is up to date.**
```
Comment (Prnyself) [inline docs/general/contributor-guide.md:54]: Please add a blank line here.

Comment (Xuanwo) [reply docs/general/contributor-guide.md:54]:
```suggestion

Notice: **Every time you try to fix an issue, you should update the upstream and checkout from the master branch to make sure your code is up to date.**
```

**Review by JinnyYi — APPROVED** [review]

**Pull request merged**